src/Tribe/Editor/Objects/Event.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Models an event (the post type) in the context of the Block Editor.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Editory\Objects
+ */
+
+namespace Tribe\Events\Editor\Objects;
+
+use Tribe__Events__Main as TEC;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class Event
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Editory\Objects
+ */
+class Event implements Editor_Object_Interface {
+	/**
+	 * The event data in the format required by the block editor.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,mixed>
+	 */
+	protected $data;
+	/**
+	 * The post object to model the data on.
+	 *
+	 * @since TBD
+	 *
+	 * @var \WP_Post
+	 */
+	protected $post;
+
+	/**
+	 * Event constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param int|\WP_Post|null $event The event post ID or object, or `null` to use the global `post` object.
+	 */
+	public function __construct( $event = null ) {
+		$event_candidate = $event ?: \tribe_get_request_var( 'post', false );
+		$this->post      = \get_post( $event_candidate );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function data( $key = null, $default = null ) {
+		if ( $this->data === null ) {
+			$this->data = [
```

Comment:
:no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: ce7ce6e342a860bad3d20d7ff7a5603e</sub>